### ISSUE TITLE
Grouping within NameValueList

### ISSUE THREAD

**Issue by jcfilben** [grommet/hpe-design-system#2906]
How should groups within a NameValueList be styled?
Question asked by Rahul in Office Hours 9/21

Some options that were brought up:
<img width="852" alt="Screen Shot 2022-09-21 at 9 25 41 AM" src="https://user-images.githubusercontent.com/54560994/191546039-16ec6d07-e729-47b2-94a8-0596302dd435.png">
<img width="692" alt="Screen Shot 2022-09-21 at 9 26 12 AM" src="https://user-images.githubusercontent.com/54560994/191546084-9c26fb9d-e669-4d17-874b-5cd83adf2770.png">
<img width="391" alt="Screen Shot 2022-09-21 at 9 26 22 AM" src="https://user-images.githubusercontent.com/54560994/191546118-83c56433-975a-4233-a4c4-0943cac07757.png">

**Comment by ericsoderberghp** [comment]
Could be somewhat related to https://github.com/grommet/hpe-design-system/issues/2899 from a visual design standpoint

**Comment by vavalos5** [comment]
@Keeth-HPe can we add a Figma link here when you can please.

**Comment by Keeth-HPe** [comment]
https://www.figma.com/file/Ny7rYK5gI5VO4PtZsq6vGC/HPE-Name-Value-List?node-id=0%3A1

I have ideated a few options but trying to understand more use cases. So that we have proper rationale/guidance behind the idea also it has to cater to all use cases.

**Comment by vavalos5** [comment]
Thank you @Keeth-HPe .

Going to invite a few folks to take a look at the different explorations so you can receive some feedback. 

@KennyAtHPE @jcfilben @taysea when you have the time today, can I get a review from you and let us know which option you like the best. Thanks!

**Comment by jcfilben** [comment]
Option 1 - I like this one the best. Feels simple and clean. 
Option 2 - My second favorite, but I wonder with this option would we keep this horizontal line separator even when we don't have grouped options within any of the sections?
Option 3 - Felt like there were too many lines on the page.
Option 4 & 5 - Too many boxes, distracts from the content.

**Comment by ericsoderberghp** [comment]
I tend to agree with Jessica. I feel like introducing horizontal rules would want to be applied in other places as well, to provide further consistency. But, we've resisted that so far in order to give a cleaner feel overall. The boxes feel counter to our background color stacking approach.

**Comment by KennyAtHPE** [comment]
I think a question to ask for this issue is "why are we grouping items in a NameValueList?" Is it to avoid visual clutter or is to differentiate and group certain fields? It's hard to tell with the use cases here since they are repeating information. 

If the reasoning is to differentiate certain fields I think all of these options do a good job of that in varying degrees for VISUAL users, however, I think that option 2 is the only option that is helpful for both visual users and users relying on voice assistive technology. The additional header here will help them understand that they are moving to a new section. My vote is for option 2, but I also do think the bolding of the header and the values on the left side are clashing.

**Comment by halocline** [comment]
I think we need the headings added for semantic structure.

I also believe this is an opportunity to use the "normal" font-weight on the H1, H2, headings which will provide visual and semantic structure, and leave the bold(er) font-weight for the Names and H3s (e.g. names of the instances). I'd also like to see a version where the H3s are normal weight as well.

**Comment by vavalos5** [comment]
> I think a question to ask for this issue is "why are we grouping items in a NameValueList?" Is it to avoid visual clutter or is to differentiate and group certain fields? It's hard to tell with the use cases here since they are repeating information.
> 
> If the reasoning is to differentiate certain fields I think all of these options do a good job of that in varying degrees for VISUAL users, however, I think that option 2 is the only option that is helpful for both visual users and users relying on voice assistive technology. The additional header here will help them understand that they are moving to a new section. My vote is for option 2, but I also do think the bolding of the header and the values on the left side are clashing.

I think you bring a valid point Kenny. I do believe the reasoning was mentioned during office hours but not captured in this ticket. @Keeth-HPe Please create a problem and problem statement. I believe you've connected with Rahul in regards to this problem, so it would be great if we could capture it. Also, revise the feedback you've received and create a second round of ideations.

**Comment by Keeth-HPe** [comment]
The problem statement :
Within the 'multiple' NameValueLists on a single page, there can be smaller groups of NameValueLists. Following the existing pattern, as per the design system guidance, does not create a significant difference between each of them. Everything looks like it belongs to one group. 

 @KennyAtHPE Thank you for bringing this up. We are basically trying to differentiate certain fields within a section of the 'Multiple NameValueList'
 @vavalos5 I've addressed the comments and made changes to the ideations.

**Comment by taysea** [comment]
I agree with @KennyAtHPE and @halocline's comments. We should use semantic structure to build the page and it seems like a fitting place to try the "normal" font weight on the heading. This solution needs to visually solve the concept of grouping and build hierarchy amongst the content in a way that still makes the data easy to read but it additionally should adhere to considerations for semantic structure.

**Comment by vavalos5** [comment]
Hi Keeth, as next steps for Monday, please continue to:

1. iterate on your challenge and problem statement. We can work on this together but give it another go.

I used the following two links to help me define my problem statement:
Hives
Scribr

2. Create four examples: 
2.1 One with using correct semantic styling. I’ve done this here in this example for you.
2.2. Second example which I also created herehttps://www.figma.com/file/Ny7rYK5gI5VO4PtZsq6vGC/HPE-Name-Value-List?node-id=2207%3A2827.
2.3 for the third example, copy and paste and utilize the same exact example i’ve created. But also play around with the name-value font sizes by bringing down the size as Matt suggested in your Figma file.
2.4 For the forth one, copy and paste it and utilize the same exact example and also play around with the name-value test sizes. Please refer to Aditi’s content she wrote in regaads to hierachy.

Thank you!

**Comment by Keeth-HPe** [comment]
Thank you all for the valuable feedback

@Vicky Thank you for all the help! I've iterated version 2 based on the feedback received. Please take a look and we shall discuss further :) 

[Figma file](https://www.figma.com/file/Ny7rYK5gI5VO4PtZsq6vGC/HPE-Name-Value-List?node-id=2224%3A4357)

**Comment by vavalos5** [comment]
Inserting Scott's comments to keep track of ticket progress: 

"Keeth and I connected this morning about [#2906](https://github.com/grommet/hpe-design-system/issues/2906).  Matt and Taylor also left comments in the associated Name-Value [figma file](https://www.figma.com/file/Ny7rYK5gI5VO4PtZsq6vGC/HPE-Name-Value-List?node-id=0%3A1). Tomorrow Keeth is out due to an Indian holiday. Thursday morning she'll resume updating the options to incorporate feedback which follow our [typography semantic structure](https://design-system.hpe.design/foundation/typography?q=hierarchy#semantic-usage-of-heading-levels) of the headings guidance.

As Taylor noted about #2906 - this ticket might require a bit more exploration into the typography hierarchy with regards to weights and sizes given in some of the stuff Chris did for Discover — We don’t want to be boxed in with the current figma typography (stylesheet) definitions. That might need to be expanded."

**Comment by Keeth-HPe** [comment]
Based on Scott's suggestion. . .(inserting the comment)  "Keeth, if our figma typography stylesheet isn't flexible enough to support where you want to take your options. Feel free to detach the text instances or create new ones, annotate the deltas."

I had the chance to Ideate and explored some more ideas. Created Version 3 where I have detached the text instances and tried using "medium and semi-bold" font weights.

https://www.figma.com/file/Ny7rYK5gI5VO4PtZsq6vGC/HPE-Name-Value-List?node-id=2316%3A3754

**Comment by SOjaHPE** [comment]
@ericsoderberghp  I've connected with @Keeth-HPe on this ticket and feel it is best that we hold off on investing any more time here since @L0ZZI is wrapping up a new typography style guide for the marketing .com web team that supports a more sophisticated typography structure not limited to just two font weights (regular and bold).

- This typography refresh work aligns w/ Gina T. initiatives, see https://github.com/grommet/grommet/issues/6461

Hopefully this web typography stylesheet will make its way into products - an updated semantically structured headers stylesheet supporting this wider weight range. Good to get alignment between authenticated and unauthenticated users.

**Comment by taysea** [comment]
Closing, stale.
